src/asymmetric_crypto/nist/private_key.rs

```diff
@@ -0,0 +1,238 @@
+use std::hash::Hash;
+
+use aead::generic_array::GenericArray;
+use elliptic_curve::{
+    sec1::{self},
+    Curve, CurveArithmetic, SecretKey,
+};
+use pkcs8::{
+    pkcs5::{pbes2, scrypt},
+    EncodePrivateKey, EncryptedPrivateKeyInfo, SecretDocument,
+};
+use rand_chacha::rand_core::CryptoRngCore;
+use rand_core::{RngCore, SeedableRng};
+use zeroize::{Zeroize, ZeroizeOnDrop, Zeroizing};
+
+#[cfg(feature = "ser")]
+use crate::bytes_ser_de::{Deserializer, Serializable, Serializer};
+use crate::{CBytes, CryptoCoreError, CsRng, FixedSizeCBytes, RandomFixedSizeCBytes, SecretCBytes};
+
+/// Nist Curve private key
+///
+/// The `LENGTH` const generic parameter is the length of the serialized private
+/// key in bytes.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct NistPrivateKey<C: Curve, const LENGTH: usize> {
+    // the secret bytes of the key
+    bytes: [u8; LENGTH],
+    // this is just a cache of the instantiated secret key
+    // to speed up curve arithmetic
+    pub(super) secret_key: SecretKey<C>,
```

Comment:
No we can't because due to the use of GenericArrays by the underlying lib, we can't recover fixed length arrays from it. (without "expecting" them).
https://github.com/Cosmian/crypto_core/issues/62